foundations/foundation-vertx/src/main/java/io/servicecomb/foundation/vertx/http/VertxServerRequestToHttpServletRequest.java

```diff
@@ -212,4 +224,83 @@ public ServletInputStream getInputStream() throws IOException {
   public AsyncContext getAsyncContext() {
     return EMPTY_ASYNC_CONTEXT;
   }
+
+  @Override
+  public Part getPart(String name) throws IOException, ServletException {
+    Optional<FileUpload> upload = context.fileUploads().stream().filter(fileUpload -> fileUpload.name().equals(name))
+        .findFirst();
+    if (!upload.isPresent()) {
+      throw new IOException("No such file with name: " + name);
+    }
+
+    CompletableFuture<Part> future = new CompletableFuture<>();
+    final FileUpload fileUpload = upload.get();
+    context.vertx().fileSystem().readFile(
+        fileUpload.uploadedFileName(),
+        fileHandler -> future.complete(new VertxPart(context, fileHandler, fileUpload)));
+    return future.join();
+  }
+
+  private static class VertxPart implements Part {
+
+    private final AsyncResult<Buffer> fileHandler;
+    private final FileUpload fileUpload;
+    private RoutingContext context;
+
+    private VertxPart(RoutingContext context, AsyncResult<Buffer> fileHandler, FileUpload fileUpload) {
+      this.fileHandler = fileHandler;
+      this.fileUpload = fileUpload;
+      this.context = context;
+    }
+
+    @Override
+    public InputStream getInputStream() throws IOException {
+      return new ByteArrayInputStream(fileHandler.result().getBytes());
```

Comment:
Is there a way to setup the max byte size that file handler can deal with?

Reply:
i didn't find a place to do that...

Reply:
any reason to setBodyLimit here?

Reply:
Just to avoid  some bad guy bomb the server with a very big attachment.

Reply:
understood, just meant why here? i set the limit somewhere else

Reply:
It's not here, we just need to make sure the limit is set.